src/extensions/users-permissions/server/services/providers-registry.js

```diff
@@ -95,19 +95,43 @@ const getInitialProviders = ({ purest }) => ({
       .auth(accessToken)
       .request();
 
-    const createdProfile = await strapi.query('api::profile.profile').create({
-      data: {
-        displayName: body.name,
-        profilePictureUrl: body.picture
-      }
+    const currentUser = await strapi.db.query('plugin::users-permissions.user').findOne({
+      where: { email: body.email }
+    });
+
+    const hasProfileAssociated = await strapi.db.query('api::profile.profile').findMany({
```

Comment:
Nit: I would call this `assoicataedProfiles`. A variable starting with `has` usually indicate it's a boolean.